packages/material-ui/src/Slider/Slider.js

```diff
@@ -430,9 +430,20 @@ const Slider = React.forwardRef(function Slider(props, ref) {
     setOpen(-1);
   });
 
+  if (disabled && active !== -1) {
+    setActive(-1);
+  }
+  if (disabled && focusVisible !== -1) {
+    setFocusVisible(-1);
+  }
+
   const isRtl = theme.direction === 'rtl';
 
   const handleKeyDown = useEventCallback((event) => {
+    if (disabled) {
+      return;
+    }
```

Comment:
This concern is still unadressed. If a disabled Slider is still focused then this should be fixed rather than ignoring keydown for these cases. If it's not focused anymore then we won't have to special case it here.

Reply:
A disabled Slider is still focused like a text box or a button would be: https://codesandbox.io/s/mui-pr-22247-forked-h151h?file=/src/App.js. Same reason why we have a disabled check in the key handler of the button base:
https://github.com/mui-org/material-ui/blob/0a84c3761b58019d99a02159c17ac5658d0cb585/packages/material-ui/src/ButtonBase/ButtonBase.js#L225

Reply:
Right this quirk again. Add a comment explaining that this is incorrect browser behavior. Disabled buttons should not be focused.

Then we can remove that branching in the test, or add appropriate expectations for each branch. One branch in the test is currently not covered by assertions.

Reply:
> Add a comment explaining that this is incorrect browser behavior. Disabled buttons should not be focused.

Are you saying it's a flaw in Firefox and Safari?

Reply:
> Elements that meet all the following criteria:
> [...]
> the element is not actually disabled;

-- https://html.spec.whatwg.org/multipage/interaction.html#focusable-area

Bug confirmed in https://bugzilla.mozilla.org/show_bug.cgi?id=1650092